src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -45,146 +73,1195 @@ ActorInit En_Slime_InitVars = {
+    stepCheck &= Math_StepToF(&this->actor.world.pos.z, this->iceBlockSnapPos.z, 10.0f);
+
+    this->actor.colorFilterTimer = 10;
+    if (stepCheck) {
+        this->actor.child =
+            Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, ACTOR_OBJ_ICEBLOCK, this->actor.world.pos.x,
+                               this->actor.world.pos.y + 30.0f, this->actor.world.pos.z, 0, 0, 0, 0);
+        if (this->actor.child != NULL) {
+            this->actor.child->cutscene = this->actor.cutscene;
+            EnSlime_SetupIceBlock(this);
+        } else {
+            this->actor.colorFilterTimer = 0;
+            this->collider.base.acFlags |= AC_ON;
+            this->iceBlockTimer = 255;
+            this->actor.flags |= ACTOR_FLAG_1;
+            this->actor.gravity = -2.0f;
+            EnSlime_SetupIdle(this);
+        }
+    }
+}
+
+/**
+ * Enter state as a chuchu frozen in an ice block.
+ * This function is called after ice block spawn is successful.
+ */
+void EnSlime_SetupIceBlock(EnSlime* this) {
+    this->actor.flags |= ACTOR_FLAG_10;
+    this->actionFunc = EnSlime_IceBlock;
+}
+
+/**
+ * Decrement the ice block timer, or if it has hit zero,
+ * check to see if the ice block child actor link has been broken
+ * or if the block has shrunk sufficiently.
+ * If timer is up and ice block is sufficiently "gone", init thaw.
+ */
+void EnSlime_IceBlock(EnSlime* this, PlayState* play) {
+    if (this->iceBlockTimer == 0) {
+        if ((this->actor.child == NULL) || (this->actor.child->update == NULL) ||
+            !(this->actor.child->scale.y >= 0.1f)) {
+            EnSlime_SetupIceBlockThaw(this);
+        }
+    } else {
+        this->actor.colorFilterTimer = 10;
+        if ((this->iceBlockTimer - 5) < 0) {
+            this->iceBlockTimer = 0;
+        } else {
+            this->iceBlockTimer -= 5;
+        }
+    }
+}
+
+/**
+ * Prepare the actor to behave as if stunned.
+ */
+void EnSlime_SetupStun(EnSlime* this) {
+    this->actor.speed = 0.0f;
+    func_800BE504(&this->actor, &this->collider);
+    if (this->actor.velocity.y > 0.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+    this->actionFunc = EnSlime_Stun;
+}
+
+/**
+ * Check the actor's timer to see whether stun has worn off.
+ * If so, either react to damage and initialize death, or start
+ * idly wobbling in a new direction.
+ */
+void EnSlime_Stun(EnSlime* this, PlayState* play) {
+    this->timer--;
+    if (this->timer == 0) {
+        EnSlime_Thaw(this, play);
+        if (this->actor.colChkInfo.health == 0) {
+            EnSlime_SetupDamaged(this, play, false);
+        } else {
+            this->actor.world.rot.y = this->actor.shape.rot.y;
+            EnSlime_SetupMoveInDirection(this);
+        }
+    }
+}
+
+/**
+ * Setup ice block thaw
+ */
+void EnSlime_SetupIceBlockThaw(EnSlime* this) {
+    this->actor.colorFilterTimer = 0;
+    this->actor.gravity = -2.0f;
+    Actor_SetColorFilter(&this->actor, 0x8000, 0x80FF, 0x2000, 10);
```

Comment:
These have defines now, which is something I wasn't aware of until I looked myself:
```suggestion
    Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_GRAY, COLORFILTER_INTENSITY_FLAG | 255, COLORFILTER_BUFFLAG_XLU, 10);
```

Hopefully I got everything right; these are brand new, so I'm pretty unfamiliar with them. I apologize in advance if it doesn't match.